src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -35,6 +37,26 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * Parses a string of {@code oneBasedIndices} into {@code Indices} and returns it.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if at least one specified index is invalid (not non-zero unsigned integer).
+     */
+    public static Indices parseIndices(String oneBasedIndices) throws ParseException {
+        String[] indices = oneBasedIndices.trim().split("\\s+");
+        int size = indices.length;
+        int[] trimmedIndices = new int[size];
+
+        for (int i = 0; i < size; i++) {
+            if (!StringUtil.isNonZeroUnsignedInteger(indices[i])) {
+                throw new ParseException(MESSAGE_INVALID_INDEX);
+            }
+            trimmedIndices[i] = Integer.parseInt(indices[i]);
+        }
+        return Indices.fromOneBased(trimmedIndices);
+    }
```

Comment:
nit: streams
this is the last stream comment I'll make, I promise :P

In particular, the below suggestion looks extremely inelegant because there is a predicate involved.
I took another look at it, and sure enough, `isNonZeroUnsignedInteger` is actually never used without a `ParseException`. Furthermore, it is never used without an accompanying `Integer.parseInt`.
This can very naturally be abstracted better, and it is my (personal) firm belief that this is what a high level of abstraction can do.

Suggestion: change `ParserUtil` to `parsePositiveIntegerElseException`, or some similar high-level function.

```
    public static Indices parseIndices(String oneBasedIndices) throws ParseException {
        Predicate<String> throwExceptionIfNotPositiveInteger = string -> {
            if (!StringUtil.isNonZeroUnsignedInteger(string)) {
                throw new RuntimeException();
            }
            return true;
        };
        int[] trimmedIndices = Arrays.stream(oneBasedIndices.trim().split("\\s+"))
                .filter(throwExceptionIfNotPositiveInteger)
                .mapToInt(Integer::parseInt)
                .toArray();
        return Indices.fromOneBased(trimmedIndices);
```

Reply:
same reason as before on exception handling